tensorflow/core/kernels/cuda_solvers.cc

```diff
@@ -202,6 +202,7 @@ Status CudaSolver::CopyLapackInfoToHostAsync(
 // numeric types.
 #define TF_CALL_LAPACK_TYPES(m) \
   m(float, S) m(double, D) m(std::complex<float>, C) m(std::complex<double>, Z)
+#define TF_CALL_LAPACK_TYPES_NO_COMPLEX(m) m(float, S) m(double, D)
```

Comment:
There is no good reason. It was done this way because it made the code slight simpler when using Eigen. But you'll notice that the Python wrapper immediately casts it back to real again:

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/linalg_ops.py#L281

We should probably fix the kernels to return real singular values in both cases. But since it requires changing the signature of the kernel, we would have to introduce a new SvdOpV2 to avoid backwards compatibility problems. Then we could get rid of the wasteful casting in the Python wrapper. I'll leave it up to you whether you are up for it in this PR. We can also do that and add complex support in a follow up.